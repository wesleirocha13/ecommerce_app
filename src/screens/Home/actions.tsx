import axios from 'axios';

export const getProducts = async ()=>{
  const response = await axios({
    method: 'get',
    url: `https://fakestoreapi.com/products`,
  })
  return response.data;
}
