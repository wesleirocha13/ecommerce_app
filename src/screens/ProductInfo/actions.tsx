import axios from 'axios';

export const getProduct = async (productId : string)=>{
  const response = await axios({
    method: 'get',
    url: `https://fakestoreapi.com/products/${productId}`,
  })
  return response.data;
}
