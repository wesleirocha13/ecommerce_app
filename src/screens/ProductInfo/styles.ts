import { StyleSheet } from "react-native";
import { primaryColor, greyColor } from '../../constants/Colors'

const styles = StyleSheet.create({
    container: {
        padding: 15,
        flex: 1,
        backgroundColor: "#FFF"
    },
    containerImg: {
        flex: 5,
    },
    containerInformationProduct:{
        flex: 2
    },
    containerDescriptionProduct: {
        overflow: "scroll"
    },
    containerPriceProduct: {
        flex: 0.7,
        justifyContent: "flex-end"
    },
    containerButtonPurchase: {
        flex: 0.9,
        justifyContent: "flex-end"
    },
    img: {
        height: "100%",
        width: "100%",
        resizeMode: "stretch",
    },
    textTitleProduct: {
        fontSize: 20,
        fontWeight: "bold",
        textAlign: "center",
        paddingTop: 5,
    },
    textDescriptionProduct: {
        paddingTop: 5,
        fontSize: 16
    },
    textPriceProduct: {
        textAlign: "center",
        fontSize: 24,
        fontWeight: "bold"
    },
    textRating: {
        textAlign: "center",
        color: greyColor,
        fontWeight: 'bold',
        fontSize: 15
    },
    buttonPurchase: {
        backgroundColor: primaryColor,
        width: '100%' ,
        height: '76%',
        borderRadius: 5,
        alignItems: 'center',
        justifyContent: 'center',
    },
    textButtonPurchase: {
        textTransform: 'uppercase',
        color: '#000',
        fontSize: 16,
        fontWeight: '700'
    },
    containerRating: {
      flexDirection: "row",
      justifyContent: "center"
    },
    containerStarRating: {
      paddingLeft: 5
    },
    containerNumberRating: {
      paddingLeft: 10
    }

});

export default styles;
