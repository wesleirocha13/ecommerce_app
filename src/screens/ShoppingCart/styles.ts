import { StyleSheet } from "react-native";
import { backgroundColor } from '../../constants/Colors'

const styles = StyleSheet.create({
    container: {
        backgroundColor: backgroundColor
    },
    containerFlatList: {
      padding: 5,
      marginTop: 5,
    },
});

export default styles;
