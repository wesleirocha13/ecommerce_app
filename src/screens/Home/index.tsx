import { View, FlatList, RefreshControl } from "react-native";
import styles from "./styles";
import { useEffect, useRef, useState } from "react";
import CardProductMedium from "../../components/CardProductMedium";
import { ProductProps } from "../../../types";
import { getProducts } from "./actions";
import { showToastError } from '../../utils/toast'

type ItemProps = {
  item: ProductProps;
};

export default function Home() {
  const [products, setProducts] = useState([]);
  const [isLoading, setIsLoading] = useState(false);

  //Obtem a lista de produtos
  const handleGetProducts = async ()=>{
    try {
      setIsLoading(true);
      const response = await getProducts();
      setProducts(response);
      setIsLoading(false);
    } catch (err) {
      showToastError('Erro ao obter os produtos. Tente novamente!');
    }
  }

  useEffect(() => {
    handleGetProducts();
  }, []);

  return (
    <View style={styles.container}>
      <FlatList
        data={products}
        keyExtractor={(item: ProductProps) => item.id.toString()}
        renderItem={({ item }: ItemProps) => <CardProductMedium item={item} />}
        contentContainerStyle={styles.containerFlatList}
        numColumns={2}
        initialNumToRender={10}
        refreshControl={
          <RefreshControl
            refreshing={isLoading}
            onRefresh={()=>handleGetProducts()}
          />
        }
      />
    </View>
  );
}
