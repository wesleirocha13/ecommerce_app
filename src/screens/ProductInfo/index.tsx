import { Text, View, Image, TouchableOpacity, ScrollView } from "react-native";
import { RouteProp } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { RootTabParamList } from "../../../types";
import { useNavigation } from "@react-navigation/native";
import styles from "./styles";

type ShoppingCartScreenProps = NativeStackNavigationProp<
  RootTabParamList,
  "ShoppingCart"
>;

type ProductInfoProps = {
  route: RouteProp<{ params: { id: string } }, "params">;
};

const data = {
  id: 1,
  name: "Produto 1",
  description:
    "Air Jordan, perfeito para jogarbasquete Tênis Nike Air Jordan, perf ênis Nike Air Jordan, perf ",
  price: 520,
  uri: "../../../assets/images/air_jordan.jpeg",
};

export default function ProductInfo({ route }: ProductInfoProps) {
  const navigation = useNavigation<ShoppingCartScreenProps>();
  return (
    <View style={styles.container}>
      <View style={styles.containerImg}>
        <Image
          source={require("../../../assets/images/air_jordan.jpeg")}
          style={styles.img}
        />
      </View>
      <View style={styles.containerInformationProduct}>
        <Text style={styles.textTitleProduct}>
          {data.name}
        </Text>
        <ScrollView style={styles.containerDescriptionProduct}>
          <Text style={styles.textDescriptionProduct}>
            {data.description}
          </Text>
        </ScrollView>
      </View>
      <View style={styles.containerPriceProduct}>
        <Text style={styles.textPriceProduct}>
          R${data.price}
        </Text>
        <Text style={styles.textPortionProduct}>
          Até 6x de {(data.price / 6).toFixed(2)} sem juros
        </Text>
      </View>
      <View style={styles.containerButtonPurchase}>
        <TouchableOpacity style={styles.buttonPurchase}>
          <Text style={styles.textButtonPurchase}>Comprar</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
}
