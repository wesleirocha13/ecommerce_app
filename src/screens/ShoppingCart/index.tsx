import { View, FlatList} from "react-native";
import CardProductSmall from "../../components/CardProductSmall";
import styles from "./styles";
import { ProductProps } from "../../../types";
import { useContext } from "react";
import CartContext from "../../context/cart";
import { CartContextProps } from "../../context/cart";
import FooterCart from './components/FooterCart'
import BlankPage from './components/BlankPage'

type ItemProps = {
  item: ProductProps;
};

export default function ShoppingCart() {
  const cartContext = useContext<CartContextProps | null>(CartContext);
  return (
    <View>
      <FlatList
        data={cartContext?.productsInCart || []}
        keyExtractor={(item: ProductProps) => item.id.toString()}
        renderItem={({ item }: ItemProps) => <CardProductSmall item={item} />}
        contentContainerStyle={styles.containerFlatList}
        numColumns={1}
        ListFooterComponent={
          cartContext?.productsInCart &&
          cartContext?.productsInCart.length > 0 ? (
            <FooterCart products={cartContext?.productsInCart || []} />
          ) : <BlankPage/>
        }
      />
    </View>
  );
}
