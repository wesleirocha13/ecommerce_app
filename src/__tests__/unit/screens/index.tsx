
import * as React from 'react';
import { render } from '@testing-library/react-native'
import ProductInfo from '../../../screens/ProductInfo'
import ShoppingCart from '../../../screens/ShoppingCart'
import Home from '../../../screens/Home'

describe('Screens snapshots', () => {
  const product:any = {id: 1, title: 'title one', description: 'description', price: 10, image: 'image', category: 'category', rating: {}}

  it('Create home screen snapshot', async () => {
    const tree = render(<Home/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Create ProductInfo screen snapshot', async () => {
    const route:any = {params: {id: '1'}}
    const tree = render(<ProductInfo route={route}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Create ShoppingCart screen snapshot', async () => {
    const tree = render(<ShoppingCart/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});

