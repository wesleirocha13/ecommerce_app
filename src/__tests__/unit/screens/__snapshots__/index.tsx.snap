// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screens snapshots Create CardProductMedium screen snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginBottom": 10,
      "opacity": 1,
      "padding": 4,
      "width": "50%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderColor": "#000",
        "borderRadius": 5,
        "borderWidth": 0.4,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "scale": 1,
          "uri": "image",
        }
      }
      style={
        Object {
          "height": 130,
          "resizeMode": "stretch",
          "width": "100%",
        }
      }
    />
    <Text
      style={
        Object {
          "maxWidth": 170,
          "overflow": "hidden",
          "paddingTop": 20,
          "textAlign": "center",
        }
      }
    >
      title one
    </Text>
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "paddingBottom": 5,
          "paddingTop": 20,
          "textAlign": "center",
        }
      }
    >
      R$
      10
    </Text>
  </View>
</View>
`;

exports[`Screens snapshots Create ProductInfo screen snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "flex": 1,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "flex": 5,
      }
    }
  >
    <Image
      source={
        Object {
          "scale": 1,
          "uri": "image",
        }
      }
      style={
        Object {
          "height": "100%",
          "resizeMode": "stretch",
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "paddingTop": 5,
          "textAlign": "center",
        }
      }
    />
    <RCTScrollView
      style={
        Object {
          "overflow": "scroll",
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "fontSize": 16,
              "paddingTop": 5,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "color": "#91939a",
          "textAlign": "center",
        }
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#a0dfdf",
          "borderRadius": 5,
          "height": "76%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 16,
            "textTransform": "uppercase",
          }
        }
      >
        Comprar
      </Text>
    </View>
  </View>
</View>
`;

exports[`Screens snapshots Create ShoppingCart screen snapshot 1`] = `
<View>
  <RCTScrollView
    ListFooterComponent={null}
    contentContainerStyle={
      Object {
        "marginTop": 5,
        "padding": 5,
      }
    }
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`Screens snapshots Create TabBarIcon screen snapshot 1`] = `<Text />`;

exports[`Screens snapshots Create home screen snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f9f9f9",
    }
  }
>
  <RCTScrollView
    ListFooterComponent={
      <ActivityIndicator
        size="large"
      />
    }
    contentContainerStyle={
      Object {
        "marginTop": 5,
        "padding": 5,
      }
    }
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.1}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <ActivityIndicator
          size="large"
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
